src/applications/static-pages/simple-forms/form-upload/App.js

```diff
@@ -0,0 +1,75 @@
+import React, { useCallback } from 'react';
+import { connect, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+import { isLoggedIn } from 'platform/user/selectors';
+import {
+  VaButton,
+  VaAlert,
+} from '@department-of-veterans-affairs/component-library/dist/react-bindings';
+import { toggleLoginModal as toggleLoginModalAction } from '@department-of-veterans-affairs/platform-site-wide/actions';
+
+export const App = ({
+  shouldDisplayFormUpload,
+  toggleLoginModal,
+  formNumber,
+  hasOnlineTool,
+}) => {
+  // TODO: Confirm this accurately detects if the user is logged in.
+  const userLoggedIn = useSelector(state => isLoggedIn(state));
+
+  const onSignInClicked = useCallback(() => toggleLoginModal(true), [
```

Comment:
Looks good, if you'd like an alternate version - also could be:
```
const dispatch = useDispatch()
const onSignInClicked = () => dispatch(toggleLoginModalAction(true))
```
I don't think callback is necessary here, unless it's used for some specific reason?
and if you use useDispatch, you'll no longer to need to use the `connect` code, which is the legacy way to do this.